test/evp_extra_test.c

```diff
@@ -1169,7 +1169,46 @@ static int test_EVP_PKCS82PKEY(void)
 
     return ret;
 }
+
 #endif
+static int test_EVP_PKCS82PKEY_wrong_tag(void)
+{
+    OSSL_PROVIDER *provider = OSSL_PROVIDER_load(NULL, "default");
```

Comment:
Why do you load the provider here? It could be the source of the failing sanitizer test problem.

Reply:
I ran into an issue when running the test suite:
```console
$ make -j8 test
...
30-test_evp.t ...................... ok    
30-test_evp_extra.t ................ 1/? 
    # ERROR: (int) 'ERR_get_error() == 0' failed @ test/evp_extra_test.c:1199
    # [503841036] compared to [0]
    # OPENSSL_TEST_RAND_ORDER=1619614042
    not ok 8 - test_EVP_PKCS82PKEY_wrong_tag
# ------------------------------------------------------------------------------
../../util/wrap.pl ../../test/evp_extra_test -context => 1
not ok 2 - running evp_extra_test with a non-default library context
30-test_evp_extra.t ................ 3/? ---------------------------------------
30-test_evp_extra.t ................ Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/3 subtests
```
which I did not get when running the test standalone:
```console
$ ./test/evp_extra_test --test test_EVP_PKCS82PKEY_wrong_tag
ok 1 - test_EVP_PKCS82PKEY_wrong_tag
```
So I added the provider here to try to work around this issue and just went by the error message mentioning using a non-default library context.

Reply:
Just add at the beginning of the test case something like:
```
if (testctx != NULL)
    /* test not supported with non-default context */
    return 1;
```

Reply:
Ah great, I'll add that. Thanks!